src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,45 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    private final BulldozerProducer bulldozerProducer = new BulldozerProducer();
+    private final TruckProducer truckProducer = new TruckProducer();
+    private final ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+
+    @Override
+    public List<Machine> getAll(Class type) {
+        List<? extends Machine> machines = null;
+        if (type == Bulldozer.class) {
```

Comment:
hint: you can use `isAssignableFrom()` method

Reply:
Can you explain, why this method differs from general equals operator, because I can't exactly understand? Why is it better than equals?))

Reply:
`==` compares references ("location" in memory) as you know
`isAssignableFrom()` is more complex - it allows you to understand whether right-side class can be assigned to a left-side class 

like `Class1.isAssignableFrom(Class2)` will return true if a variable of type `Class2` can be assigned to a variable of type `Class1`

in our case we indeed can use `==` , since we want to check an exact match

therefore I would consider my previous comment as unnecessary, since I missed that we need an exact match only :)